modules/db/src/main/java/org/mule/module/db/internal/domain/param/DefaultInputQueryParam.java

```diff
@@ -14,20 +14,38 @@
 
     private final Object value;
 
+    private final boolean isDbParam;
+
     public DefaultInputQueryParam(int index, DbType type, Object value)
     {
-        this(index, type, value, null);
+        this(index, type, value, null, false);
     }
 
     public DefaultInputQueryParam(int index, DbType type, Object value, String name)
     {
+        this(index, type, value, name, false);
+    }
+
+    public DefaultInputQueryParam(int index, DbType type, Object value, String name, boolean isDbParam)
+    {
         super(index, type, name);
         this.value = value;
+        this.isDbParam = isDbParam;
     }
 
     @Override
     public Object getValue()
     {
         return value;
     }
+
+    /**
+     * @return true if the InputQueryParam references a param that must be defined through a <db:in-param> element in the configuration file.
```

Comment:
Rename the method as mentioned before. Javadoc must say something like "indicates if the parameter has a defined value...... This is useful when a paremter is defined in a query template...."